### PR TITLE
CR-263 tmixture functions

### PR DESCRIPTION
**Pull request by EpigeneMax**
## Description
See [Jira card](https://epigenelabsteam.atlassian.net/browse/CR-263)

## Motivation and Context
Splitting the port of `limma` into smaller PR to ease the integration process

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] No code (non-breaking change that does not impact code: formatting, build system, documentation...)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist

- [x] My changes are in a separate branch named after a Jira card, or -- if not applicable -- with a descriptive name.
- [x] I have descriptive commit messages, with a short title (first line).
- [x] I have added and/or updated tests to reflect my changes.
- [x] I have reviewed and updated the documentation accordingly.
- [x] I have updated the [Changelog](../CHANGELOG.md).
- [x] My code and documentation follow the [numpy docstring format](https://sphinxcontrib-napoleon.readthedocs.io/en/latest/example_numpy.html).
- [x] I have reviewed possible licensing issues, preferably with a person in charge of the topic.

### CODE REVIEW

File: inmoose/limma/ebayes.py [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+# -----------------------------------------------------------------------------
+
+# This file is based on the file 'R/ebayes.R' of the Bioconductor limma package (version 3.55.1).
+
+
+import numpy as np
+import scipy
+
+
+def tmixture_matrix(tstat, stdev_unscaled, df, proportion, v0_lim=None):
```
Comment (soleneweill) [inline inmoose/limma/ebayes.py:26]: Could you add variable type ?

File: inmoose/limma/ebayes.py [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+    v0 = np.array(
+        [
+            tmixture_vector(tstat[:, j], stdev_unscaled[:, j], df, proportion, v0_lim)
+            for j in range(ncoef)
+        ]
+    )
+    return v0
+
+
+def tmixture_vector(tstat, stdev_unscaled, df, proportion, v0_lim=None):
```
Comment (soleneweill) [inline inmoose/limma/ebayes.py:77]: Same here can you add variable type ?

**Review by soleneweill — APPROVED** [review]
Looks good, only comments on the docstring :)

File: inmoose/limma/ebayes.py (hunk 1 above)
Comment (EpigeneMax) [reply inmoose/limma/ebayes.py:26]: types are documented in the docstring
I'm not sure that type annotation would add much info

File: inmoose/limma/ebayes.py (hunk 2 above)
Comment (EpigeneMax) [reply inmoose/limma/ebayes.py:77]: types are documented in the docstring
I'm not sure that type annotation would add much info

File: inmoose/limma/ebayes.py (hunk 1 above)
Comment (soleneweill) [reply inmoose/limma/ebayes.py:26]: It's just that when you use it outside and you have the type in the function definition you can see it easily. If you think it is not worth it I trust you

**Pull request merged**